### PR TITLE
Clean legacy SimpleMDE code

### PR DESCRIPTION
**Pull request by wxiaoguang**
Since we are using EasyMDE now, we do not need to keep the SimpleMDE code anymore.

This PR removes all legacy SimpleMDE code, and makes some related changes:
* `createCommentEasyMDE` can accept native DOM element, and it doesn't need `jQuery.data` to store EasyMDE editor object (as discussed about the frontend guideline).
* introduce `getAttachedEasyMDE` to get the attached EasyMDE editor object, it's easier to find all the usage of EasyMDE.
* rename variable names from `$simplemde` to `easyMDE`, the `$` was incorrect because it is a EasyMDE editor, not a jQuery object.

With this PR, it will be easier to do more refactoring or replacing EasyMDE with other editors.

### CODE REVIEW

File: routers/web/repo/commit.go [hunk 1]
```diff
@@ -252,7 +252,7 @@ func FileHistory(ctx *context.Context) {
 func Diff(ctx *context.Context) {
 	ctx.Data["PageIsDiff"] = true
 	ctx.Data["RequireHighlightJS"] = true
-	ctx.Data["RequireSimpleMDE"] = true
+	ctx.Data["RequireEasyMDE"] = true
```
Comment (silverwind) [inline routers/web/repo/commit.go:256]: Can we do away with this variable and just detect the presence of certain DOM elements in JS instead?

Comment (wxiaoguang) [reply routers/web/repo/commit.go:256]: Good idea!

Maybe async loading should work, or we could get this merged if it isn't worse than before 😊

Comment (wxiaoguang) [reply routers/web/repo/commit.go:256]: @silverwind 

Could we move on? I think the refactoring of `RequireEasyMDE` can be done in a separate PR. This PR has done enough. Removing `RequireEasyMDE` might involve unrelated changes.

**Review by KN4CK3R — APPROVED** [review]

File: web_src/js/features/comp/CommentEasyMDE.js [hunk 2]
```diff
@@ -65,8 +74,25 @@ export function createCommentSimpleMDE($editArea) {
...
+ * get the attached EasyMDE editor created by createCommentEasyMDE
+ * @param el jQuery or HTMLElement
+ * @returns {null|EasyMDE}
+ */
+export function getAttachedEasyMDE(el) {
+  if (el instanceof jQuery) {
+    el = el[0];
+  }
+  if (!el) {
+    return null;
```
Comment (silverwind) [inline web_src/js/features/comp/CommentEasyMDE.js:95]:
```suggestion
    return el;
```

Make the function more transparent.

File: web_src/js/features/repo-issue.js [hunk 3]
```diff
@@ -214,8 +214,8 @@ export function initRepoIssueStatusButton() {
   // Change status
   const $statusButton = $('#status-button');
   $('#comment-form textarea').on('keyup', function () {
-    const $simplemde = $(this).data('simplemde');
-    const value = ($simplemde && $simplemde.value()) ? $simplemde.value() : $(this).val();
+    const easyMDE = getAttachedEasyMDE(this);
+    const value = (easyMDE && easyMDE.value()) ? easyMDE.value() : $(this).val();
```
Comment (silverwind) [inline web_src/js/features/repo-issue.js]:
```suggestion
    const value = easyMDE?.value() || $(this).val();
```

**Review by silverwind — APPROVED** [review]

**Comment by wxiaoguang** [comment]
trigger bot

**Pull request merged**